### PR TITLE
(WIP) Manage project groups

### PR DESCRIPTION
**Pull request by ningt**
Add project groups controller and basic views

### Tasks
 - [ ] :+1: from team

### References
 - Jira link:

### Risks
- Level: Medium

### CODE REVIEW

File: app/views/shared/_header.html.erb [hunk 1]
```diff
@@ -65,6 +77,7 @@
             <li><%= link_to "Environments", environments_path %></li>
             <li><%= link_to "Deploy Groups", deploy_groups_path %></li>
           <% end %>
+          <li><%= link_to "Project Groups", project_groups_path %></li>
```
Comment (baopham) [inline app/views/shared/_header.html.erb]: note: only show if `ProjectGroup.enabled`

File: app/views/project_groups/_fields.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  <%= form.input :permalink, required: true if form.object.persisted? %>
+
+  <% if SecretStorage.backend == Samson::Secrets::HashicorpVaultBackend %>
+    <%= form.input :vault_server_id do %>
+      <%= form.collection_select(
+            :vault_server_id,
+            Samson::Secrets::VaultServer.select(:id, :name).all, :id, :name,
+            {include_blank: true}, { class: 'form-control' })
+      %>
+    <% end %>
```
Comment (baopham) [inline app/views/project_groups/_fields.html.erb]: note: update the fields:

* Name
* Description (optional)

File: app/controllers/project_groups_controller.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    private
+
+    def project_group_params
+        params.require(:project_group).permit(*allowed_project_group_params)
+    end
+
+    def allowed_project_group_params
+        (
+            [:name, :vault_server_id, :permalink]
```
Comment (baopham) [inline app/controllers/project_groups_controller.rb]: TODO: remove copy&paste code `vault_server_id`

File: app/controllers/project_groups_controller.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
+class ProjectGroupsController < ApplicationController
+    before_action :authorize_super_admin!, except: [:index, :show]
+    before_action :project_group, only: [:edit]
+
+    def index
+        @project_groups = ProjectGroup.where(nil)
```
Comment (baopham) [inline app/controllers/project_groups_controller.rb]: can just do `ProjectGroup.all`?

File: app/controllers/project_groups_controller.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
+class ProjectGroupsController < ApplicationController
```
Comment (baopham) [inline app/controllers/project_groups_controller.rb]: Use 2-spaces indentation please

File: app/views/project_groups/_fields.html.erb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<fieldset>
+  <%= form.input :name, required: true %>
+  <%= form.input :permalink, required: true if form.object.persisted? %>
+  </fieldset>
```
Comment (timotheeg) [inline app/views/project_groups/_fields.html.erb]: unnecessary space

**Pull request merged**